tests/relay/schema.py

```diff
@@ -0,0 +1,209 @@
+from typing import Iterable, List, Optional
+from typing_extensions import Self
+
+import strawberry
+from strawberry.relay.utils import to_base64
+from strawberry.types import Info
+
+
+@strawberry.type
+class Fruit(strawberry.relay.Node):
+    _id: strawberry.Private[int]
+    name: str
+    color: str
+
+    @classmethod
+    def resolve_id(cls, root: Self, *, info: Optional[Info] = None):
+        return root._id
+
+    @classmethod
+    def resolve_nodes(
+        cls,
+        *,
+        info: Optional[Info] = None,
+        node_ids: Optional[Iterable[str]] = None,
```

Comment:
when would get get no `node_ids`?

Reply:
The main situation is: The `NodeField` have `id` or `ids` depending on whether it is a list or not. For example:

```python
@strawberry.type
class Query:
    node: Node
    nodes: List[Node]
```

This will generate:

```graphql
type Query {
  node(id: GlobalID!): Node!
  nodes(ids: [GlobalID!]!): [Node!]!
}
```

When querying `nodes` the list of global IDs will be passed in there

But it can be useful as a utility function for end users too.

Reply:
Oh, actually you question is when we would "not" get those.

When resolving a connection for example. The connection asks the `Node` for his nodes and paginate them. In this situation there isn't a list of nodes to be filtered, it just wants an iterable of "everything" (not actually everything, usually will be a lazy object that can be sliced) to slice.